dill/_dill.py

```diff
@@ -1665,60 +1687,78 @@ def save_classmethod(pickler, obj):
+                    getattr(_import_module(obj.__module__, True), '__dict__', None) is globs_copy:
+                globs = globs_copy
             else:
-                postproc_list.append((setattr, (obj, '__dict__', obj.__dict__)))
-                members = None
-            if id(globs) not in pickler.memo:
-                globs_old = globs
                 globs = {'__name__': obj.__module__}
-                if type(globs_old) is not dict:
-                    globs = dict.__new__(type(globs_old), globs)
-                # Code from Python 2 cannot be transfered to Python 3 anyway because of bytecode incompatibility, so this good
-                postproc_list.append((dict.update, (globs, globs_old)))
-        else:
-            members = obj.__dict__
+
+        if globs_copy is not None and globs is not globs_copy:
+            # In the case that the globals are copied, we need to ensure that
+            # the globals dictionary is updated when all objects in the
+            # dictionary are already created.
+            if PY3:
+                glob_ids = {id(g) for g in globs_copy.values()}
+            else:
+                glob_ids = {id(g) for g in globs_copy.itervalues()}
+            for stack_element in _postproc:
+                if stack_element in glob_ids:
+                    _postproc[stack_element].append((dict.update, (globs, globs_copy)))
+                    break
+            else:
+                postproc_list.append((dict.update, (globs, globs_copy)))
+
         if PY3:
-            #NOTE: workaround for 'super' (see issue #75)
-            _super = ('super' in getattr(obj.__code__,'co_names',())) and (_byref is not None)
+            #NOTE: workaround for 'super' (see issue #75) removed in #443
             fkwdefaults = getattr(obj, '__kwdefaults__', None)
             _save_with_postproc(pickler, (_create_function, (
                   obj.__code__, globs, obj.__name__, obj.__defaults__,
-                  obj.__closure__, members, fkwdefaults
-            )), obj=obj, postproc_list=postproc_list) #, delayed_objs=(globs)
+                  obj.__closure__, obj.__dict__, fkwdefaults
+            )), obj=obj, postproc_list=postproc_list)
         else:
-            _super = ('super' in getattr(obj.func_code,'co_names',())) and (_byref is not None) and getattr(pickler, '_recurse', False)
             _save_with_postproc(pickler, (_create_function, (
                 obj.func_code, globs, obj.func_name, obj.func_defaults,
-                obj.func_closure, members
+                obj.func_closure, obj.__dict__
             )), obj=obj, postproc_list=postproc_list)
-       #clear = (_byref, _super, _recurse, _memo)
-       #print(clear + (OLDER,))
-        #NOTE: workaround for #234; "partial" still is problematic for recurse
-        if OLDER and not _byref and (_super or (not _super and _memo) or (not _super and not _memo and _recurse)): pickler.clear_memo()
```

Comment:
This is all very good... and fixes a dirty hack.  So, I hate to ask this question.  Have you looked at the impact of this change on older pickles that use the `_super` hack?

Reply:
@anivegesana: ?